scio-core/src/main/java/com/spotify/scio/transforms/FutureHandlers.java

```diff
@@ -45,7 +45,7 @@ public class FutureHandlers {
     default void waitForFutures(Iterable<ListenableFuture<V>> futures)
         throws InterruptedException, ExecutionException {
       // Futures#allAsList only works if all futures succeed
-      Futures.whenAllComplete(futures).run(() -> {}, MoreExecutors.directExecutor()).get();
+      Futures.successfulAsList(futures).get();
```

Comment:
just wondering, does this use the same executor (`MoreExecutors.directExecutor()`)? 

Also, how does `successfulAsList` handle failed futures? does it await all futures and only return failed ones?